packages/cletus/src/operations/clerk.tsx

```diff
@@ -665,6 +666,158 @@ export const file_read = operationOf<
   ),
 });
 
+export const file_edit = operationOf<
+  { path: string; request: string; offset?: number; limit?: number },
+  { oldContent: string; newContent: string; changed: boolean; diff: string }
+>({
+  mode: 'update',
+  signature: 'file_edit(path: string, request: string, offset?: number, limit?: number)',
+  status: (input) => `Editing: ${path.basename(input.path)}`,
+  analyze: async (input, { cwd }) => {
+    const fullPath = path.resolve(cwd, input.path);
+    const readable = await fileIsReadable(fullPath);
+
+    if (!readable) {
+      return {
+        analysis: `This would fail - file "${input.path}" not found or not readable.`,
+        doable: false,
+      };
+    }
+
+    const writable = await fileIsWritable(fullPath);
+    if (!writable) {
+      return {
+        analysis: `This would fail - file "${input.path}" is not writable.`,
+        doable: false,
+      };
+    }
+
+    const type = await categorizeFile(fullPath, input.path);
+    if (type !== 'text') {
+      return {
+        analysis: `This would fail - file "${input.path}" is not a text file (type: ${type}).`,
+        doable: false,
+      };
+    }
+
```

Comment:
Do diff computation in here as well. analysis is the patch. Move the logic outside of do into a reusable function like what's seen in architect.tsx type_update tool (validation function).